db/20230420210648_batch-aggregation-state.up.sql

```diff
@@ -0,0 +1,10 @@
+-- We create a new state for batch_aggregations.
+-- AggregateShare is now optional (if there are no reports), so we drop the NOT NULL constraint.
+
+CREATE TYPE BATCH_AGGREGATION_STATE AS ENUM(
+    'AGGREGATING',  -- this batch aggregation has not been collected & permits further aggregation
+    'COLLECTED'     -- this batch aggregation has been collected & no longer permits aggregation
+);
+
+ALTER TABLE batch_aggregations ADD COLUMN state BATCH_AGGREGATION_STATE NOT NULL;  -- the current state of this batch aggregation
```

Comment:
Hmm... I don't think it makes a difference.

Without `DEFAULT 'AGGREGATING'` (or some other default value), this migration can't be applied if there are any `batch_aggregations` rows since we would be trying to create a new, non-nullable column with no default value.

But `DEFAULT 'AGGREGATING'` is not correct for a system which already has batch aggregations; for correctness, we would need already-collected aggregations to move to state 'COLLECTED'. But we'd need to implement a data migration to make this happen.

Since we are currently not implementing data migrations for non-backwards-compatible schema changes, I'd prefer not specifying a default value: the main application code doesn't need it; it might hypothetically mask bugs in application code; and the lack of a default can be viewed as providing a safeguard against applying this migration to a system with data, since it will fail.